KOBUMANZI TRISHIA<br>
M24B23/011<br>
B27505  BSCS

Import Libraries

In [26]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

Importing Healthcare dataset as pandas dataframe called Healthcare_data

In [27]:
Healthcare_data=pd.read_csv("healthcare_dataset.csv")
Healthcare_data.head()

Name  Age  Gender Blood Type Medical Condition Date of Admission  \
0  Bobby JacksOn   30    Male         B-            Cancer        2024-01-31   
1   LesLie TErRy   62    Male         A+           Obesity        2019-08-20   
2    DaNnY sMitH   76  Female         A-           Obesity        2022-09-22   
3   andrEw waTtS   28  Female         O+          Diabetes        2020-11-18   
4  adrIENNE bEll   43  Female        AB+            Cancer        2022-09-19   

             Doctor                    Hospital Insurance Provider  \
0     Matthew Smith             Sons and Miller         Blue Cross   
1   Samantha Davies                     Kim Inc           Medicare   
2  Tiffany Mitchell                    Cook PLC              Aetna   
3       Kevin Wells  Hernandez Rogers and Vang,           Medicare   
4    Kathleen Hanna                 White-White              Aetna   

   Billing Amount  Room Number Admission Type Discharge Date   Medication  \
0    18856.281306          328         Urgent     2024-02-02  Paracetamol   
1    33643.327287          265      Emergency     2019-08-26    Ibuprofen   
2    27955.096079          205      Emergency     2022-10-07      Aspirin   
3    37909.782410          450       Elective     2020-12-18    Ibuprofen   
4    14238.317814          458         Urgent     2022-10-09   Penicillin   

   Test Results  
0        Normal  
1  Inconclusive  
2        Normal  
3      Abnormal  
4      Abnormal

In [28]:
#Checking for the data types
Healthcare_data.dtypes

Name                   object
Age                     int64
Gender                 object
Blood Type             object
Medical Condition      object
Date of Admission      object
Doctor                 object
Hospital               object
Insurance Provider     object
Billing Amount        float64
Room Number             int64
Admission Type         object
Discharge Date         object
Medication             object
Test Results           object
dtype: object

### a).Explain the central tendency and distribution of any two continuous variables<br>
    

Using central tendency stats to explain central tendency of Age and Billing Amount

In [29]:
Healthcare_data.describe()

Age  Billing Amount   Room Number
count  55500.000000    55500.000000  55500.000000
mean      51.539459    25539.316097    301.134829
std       19.602454    14211.454431    115.243069
min       13.000000    -2008.492140    101.000000
25%       35.000000    13241.224652    202.000000
50%       52.000000    25538.069376    302.000000
75%       68.000000    37820.508436    401.000000
max       89.000000    52764.276736    500.000000

            Explanation:
* The central tendency of Age (mean =51.539459 , median =52.000000) shows that median is greater than mean which implies that age is normally distributed. This could result from their being a limit on the maximum age of a patient but no limit on the minimum age.

* The central tendency of Billing Amount (mean =25539.316097, median =25538.069376 ) shows that mean is greter than median which implies that it is not normally distributed implying that there's a limit on the minimum Billing amount charged but no limit on the maximum charged. 

Explaining distribution of Age and Billing Amount using correlation

In [30]:
Grouped_data=Healthcare_data[["Age","Billing Amount"]]
Grouped_data.corr()


Age  Billing Amount
Age             1.000000       -0.003832
Billing Amount -0.003832        1.000000

* There is a very weak negative correlation between Age and Billing Amount.<br>
This suggests that there is almost no linear relationship between the two variables therefore age does not have a significant impact on the billing amount.

### b).Explain the marginal probability of any two categorical variables.

    Marginal probability of Gender

In [31]:

Gender_probability = Healthcare_data['Gender'].value_counts()/len(Healthcare_data)
print("The marginal probabilities for Gender are: \n",Gender_probability)

The marginal probabilities for Gender are: 
 Gender
Male      0.500432
Female    0.499568
Name: count, dtype: float64


     Explanation
* The probability that patients are male is higher (50.0432%) than female (49.9568%).

     Marginal probability of Blood type

In [32]:
Bloodtype_probability = Healthcare_data['Blood Type'].value_counts()/len(Healthcare_data)
print("The marginal probabilities for Blood type are: \n",Bloodtype_probability)

The marginal probabilities for Blood type are: 
 Blood Type
A-     0.125568
A+     0.125333
AB+    0.125171
AB-    0.125135
B+     0.125135
B-     0.125117
O+     0.124631
O-     0.123910
Name: count, dtype: float64


* Patients with A- have the highest probability(12.5568%) and those with O- have the lowest(12.391%).

### c).Explain the relationship between the "Medical condition" and "Admission Type"

In [33]:
#Generating a cross-tabulation between 'Medical condition' and 'Admission Type'
Crosstable=pd.crosstab(index=Healthcare_data['Medical Condition'],columns=Healthcare_data['Admission Type'])

#Chi- square test for Crosstable(Medical condition and Admission Type)
from scipy.stats import chi2_contingency
Chi_square_result = chi2_contingency(Crosstable)
print(f"The P-value is {Chi_square_result[1]}")

The P-value is 0.036171783236322336


* Since the P-value is less than 0.05,Medical Condition and Admission Type are correlated.